src/download-pool.js

```diff
@@ -2,38 +2,75 @@ import downloadFile from './download-file';
 
 const POOL_LIMIT = 6;
 
+/**
+ * Groups the elements in an array in subarrays defined by limit.
+ * @example
+ *
+ * splice([1, 2, 3, 4, 5], 3); // [[1, 2, 3], [4, 5]]
+ *
+ * @param {Array} arr
+ * @param {Number} limit
+ * @returns {Array} New array with the subgroups.
+ */
 export const splice = (arr, limit) => {
   const splicedArray = [];
+  const baseArray = arr.slice(0);
```

Comment:
Good! I didn't know that, but I think any of them is really clear. It should be better to create a function `cloneArray` and in that way, it will be transparent how does it work internally. I like to use that kind of helper functions, since they make the code cleaner for anyone, but I don't know if you guys prefer to use the native code directly. Another example is `fs.stat` in downloadFile. It doesn't say anything and a comment is needed in order to explain what's happening. A helper function will help there too. If you agree @alejandronanez and @andresdavid90, I can create a helpers file where we can have these kind of helpers.